src/RepositoryChainsArchiver.ts

```diff
@@ -209,80 +219,33 @@ export default class RepositoryChainsArchiver {
     } while (continuationToken);
   }
 
-  async uploadS3(s3Config: S3Config): Promise<void> {
-    const s3Client = new S3Client({
-      region: s3Config.region,
-      credentials: {
-        accessKeyId: s3Config.accessKeyId,
-        secretAccessKey: s3Config.secretAccessKey,
-      },
-      endpoint: s3Config.endpoint,
-    });
-    try {
-      const timestamp = new Date()
-        .toISOString()
-        .replace(/[:]/g, "-")
-        .split(".")[0];
-      const backupName = `sourcify-repository-${timestamp}`;
-
-      // Generate manifest
-      const manifest = {
-        description:
-          "Manifest file for when the Sourcify file repository was uploaded",
-        timestamp: Date.now(),
-        dateStr: new Date().toISOString(),
-        files: Object.keys(this.tarStreams).map((archiveName) => ({
-          path: `${backupName}/${archiveName}`,
-          sizeInBytes: fs.statSync(path.join(this.exportPath, archiveName))
-            .size,
-        })),
-      };
-
-      // Write manifest locally
-      const manifestPath = path.join(this.exportPath, "manifest.json");
-      await fs.promises.writeFile(
-        manifestPath,
-        JSON.stringify(manifest, null, 2)
-      );
-
-      // Upload all tar files
-      for (const archiveName of Object.keys(this.tarStreams)) {
-        const localPath = path.join(this.exportPath, archiveName);
-        const s3Key = `${backupName}/${archiveName}`;
-        await this.uploadFile(s3Client, s3Config.bucketName, localPath, s3Key);
-      }
-
-      // Upload manifest files
-      await this.uploadFile(
-        s3Client,
-        s3Config.bucketName,
-        manifestPath,
-        `${backupName}/manifest.json`
-      );
-      await this.uploadFile(
-        s3Client,
-        s3Config.bucketName,
-        manifestPath,
-        "manifest.json"
-      );
+  async uploadManifest(
+    backupName: string,
+    uploadedFiles: { path: string; sizeInBytes: number }[]
+  ): Promise<void> {
+    // Generate manifest
+    const manifest = {
+      description:
+        "Manifest file for when the Sourcify file repository was uploaded",
+      timestamp: Date.now(),
+      dateStr: new Date().toISOString(),
+      files: uploadedFiles,
+    };
+
+    // Write manifest locally
+    const manifestPath = path.join(this.exportPath, "manifest.json");
+    await fs.promises.writeFile(
+      manifestPath,
+      JSON.stringify(manifest, null, 2)
+    );
 
-      // Clean up old backups
-      await this.deleteOldBackups(s3Client, s3Config.bucketName);
+    // Upload manifest files
+    await this.uploadFile(manifestPath, `${backupName}/manifest.json`);
+    await this.uploadFile(manifestPath, "manifest.json");
```

Comment:
Why are there two uploads?

Reply:
The bucket currently has this structure:
```
sourcify-repository-2024-12-17T00-00-21/
    -- manifest.json
    -- [other files]
sourcify-repository-2024-12-22T00-00-20/
    -- manifest.json
    -- [other files]
sourcify-repository-2024-12-23T00-00-26/
    -- manifest.json
    -- [other files]
manifest.json
```
So the manifest is duplicated, because we keep 3 backups. the manifest.json on the highest level of the bucket refers to the newest backup always. Let me know if you think anything of this should be changed.

Reply:
Is the root manifest.json exactly the same as <latest-date>/manifest.json? Or is the top manifest a manifest of manifests? 

IMO we should redirect [repo-backup.sourcify.dev](https://repo-backup.sourcify.dev/) to the latest manifest, meaning root and latest should be duplicates. But we should document that the prev 2 backups are also available and can be seen with `ls`.

Reply:
yes they are exactly the same

Reply:
added a task to https://github.com/ethereum/sourcify/issues/1695

Reply:
A small thing then, if they are exactly the same, is to make the paths absolute in manifest.json:

```js
    {
      // "path": "sourcify-repository-2024-12-23T03-00-23/sourcify-repository-2024-12-23T03-00-23.part.gz.aa",
      "path": "/sourcify-repository-2024-12-23T03-00-23/sourcify-repository-2024-12-23T03-00-23.part.gz.aa",
      "sizeInBytes": 2097152000
    },
```

With that both manifests will lead to the same full URL instead of potentially interpreting this as relative paths.

https://repo-backup.sourcify.app/sourcify-repository-2024-12-23T03-00-23/sourcify-repository-2024-12-23T03-00-23.part.gz.aa

Reply:
ah yes that should be fixed